lib/chef/provider/package/windows.rb

```diff
@@ -129,6 +129,7 @@ def action_install
         # Chef::Provider::Package action_install + action_remove call install_package + remove_package
         # Pass those calls to the correct sub-provider
         def install_package(name, version)
+          fail Chef::Exceptions::NoWindowsPackageSource, "installer_type '#{installer_type}' requires a source" if new_resource.source.nil?
```

Comment:
If you look at some of the other providers you'll see that they have assertions that also depend on values that come from load_current_resource, which in this case contains a call to package_provider, so we should know by then.